src/Behat/Behat/Context/ContextFactory.php

```diff
@@ -43,6 +48,7 @@
     public function __construct(ArgumentOrganiser $argumentOrganiser)
     {
         $this->argumentOrganiser = $argumentOrganiser;
+        $this->validator = new Validator();
```

Comment:
shouldn't you inject this as constructor argument ?

Reply:
No. I don't envision any other implementation of Validator at the moment. I also don't see it becoming dependent on any external service.

In this regard there's no real difference between

```php
public function __construct(...)
{
    $this->validator = new Validator();
}
```

and

```php
public function __construct(..., Validator $validator = null)
{
    $this->validator = $validator ?: new Validator();
}
```

I'd rather not expose any extra dependencies without good reason yet.